File: src/modifiers/accessibilityActions.ts
import type { ReactNode } from 'react';
import { createModifier } from './createModifier';

export interface AccessibilityActionsParams {
  /// A view (typically `Button`s) declaring the actions available to
  /// assistive technologies for this element.
  content: ReactNode;
}

/// SwiftUI `.accessibilityActions(_:)` (watchOS 10+). Declares a set of
/// custom accessibility actions from the provided content. On watchOS 9 the
/// modifier is a no-op and the view is returned unchanged.
export function accessibilityActions(params: AccessibilityActionsParams) {
  return createModifier('accessibilityActions', params);
}

File: apple/Sources/ReactNativeWatchOS/Modifiers/AccessibilityActions.swift
import SwiftUI

/// SwiftUI `.accessibilityActions(_:)` (watchOS 10+). Declares a set of
/// custom accessibility actions from the content slot. On watchOS 9 the
/// view is returned unchanged.
enum RNWAccessibilityActionsModifier {
    @MainActor
    static func register(into r: RNWModifierRegistry) {
        r.register("accessibilityActions") { view, params, ctx in
            guard let body = ctx.content(params.string("content")) else { return view }
            if #available(watchOS 10.0, *) {
                return AnyView(view.accessibilityActions { body })
            }
            return view
        }
    }
}
